File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        quer = []
        current = ""
        for unit in s:
            if unit == "{" or unit == "(" or unit == "[":
                quer.append(unit)
                current = unit
            else:
                if current and quer:
                    if quer[-1] == "{" and unit != "}":
                        return False
                    elif quer[-1] == "[" and unit != "]":
                        return False
                    elif quer[-1] == "(" and unit != ")":
                        return False
                    del quer[-1]
                    if len(quer) > 0:
                        current = quer[-1]
                else:
                    return False
        if quer:
            return False
        return True

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<String> stacker = new Stack<String>();
        
        if(s.length() == 0 || s.length()%2 != 0){
            return false;
        }
        
        for(int i = 0; i < s.length(); i++){
            String current = "";
            current += s.charAt(i);
            if(current.equals(")"))
            {
                if(stacker.isEmpty()){
                    return false;
                }
                String value = stacker.pop();
                if(!value.equals("(")){
                    return false;
                }
            }
            else if(current.equals("}"))
            {
                if(stacker.isEmpty()){
                    return false;
                }
                String value = stacker.pop();
                if(!value.equals("{")){
                    return false;
                }
            }
            else if(current.equals("]"))
            {
                if(stacker.isEmpty()){
                    return false;
                }
                String value = stacker.pop();
                if(!value.equals("[")){
                    return false;
                }
            }
            else{
                
                stacker.push(current);
                System.out.println(stacker);
            }
        }
        if(stacker.isEmpty()){
        return true;
        }
        else{
            return false;
        }
    }
}